### PR TITLE
[rc-swift] Personalization

### PR DESCRIPTION
**Pull request by paulb777**
#no-changelog

### CODE REVIEW

File: FirebaseRemoteConfig/Sources/FIRRemoteConfig.m [hunk 1]
```diff
@@ -182,7 +181,7 @@ - (instancetype)initWithAppName:(NSString *)appName
       RCNPersonalization *personalization =
           [[RCNPersonalization alloc] initWithAnalytics:analytics];
```
Comment (ncooke3) [inline FirebaseRemoteConfig/Sources/FIRRemoteConfig.m:182]: fyi, these warnings were introduced by me and will be addressed when migrating Analytics interop to its own pod.

File: FirebaseRemoteConfig/SwiftNew/Personalization.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+private let kExternalGroupParam = "group"
+
+private let kInternalEvent = "_fpc"
+private let kChoiceId = "choiceId"
+private let kInternalChoiceIdParam = "_fpid"
+
+@objc(RCNPersonalization)
+public class Personalization: NSObject {
+  /// Analytics connector.
+  weak var analytics: FIRAnalyticsInterop?
```
Comment (ncooke3) [inline FirebaseRemoteConfig/SwiftNew/Personalization.swift:38]: Note the Obj-C property is strong.

```objc
/// Analytics connector
@property(nonatomic, strong) id<FIRAnalyticsInterop> _Nullable analytics;
```

File: FirebaseRemoteConfig/SwiftNew/Personalization.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  /// Called when an arm is pulled from Remote Config. If the arm is personalized, log information
+  /// to
+  /// Google Analytics in another thread.
+  @objc public func logArmActive(rcParameter: String, config: [String: Any]) {
+    guard let ids =
+      config[ConfigConstants.fetchResponseKeyPersonalizationMetadata] as? [String: Any],
+      let values = config[ConfigConstants.fetchResponseKeyEntries] as? [String: RemoteConfigValue],
+      let value = values[rcParameter] else {
+      return
+    }
```
Comment (ncooke3) [inline FirebaseRemoteConfig/SwiftNew/Personalization.swift:57]: I don't know if this makes a difference in practice but the Obj-C impl. would early exit for non-nil, empty dictionaries. Adding the count check:
```suggestion
    guard let ids =
      config[ConfigConstants.fetchResponseKeyPersonalizationMetadata] as? [String: Any],
      !ids.isEmpty,
      let values = config[ConfigConstants.fetchResponseKeyEntries] as? [String: RemoteConfigValue],
      !values.isEmpty,
      let value = values[rcParameter] else {
      return
    }
```

**Review by ncooke3 — APPROVED** [review]

File: FirebaseRemoteConfig/SwiftNew/Personalization.swift (hunk 2 above)
Comment (paulb777) [reply FirebaseRemoteConfig/SwiftNew/Personalization.swift]: Good catch! Removed `weak`.

File: FirebaseRemoteConfig/SwiftNew/Personalization.swift (hunk 3 above)
Comment (paulb777) [reply FirebaseRemoteConfig/SwiftNew/Personalization.swift:57]: It's a redundant check since the following subscript check subsumes the empty check.

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**